[TF2] Add client predicted autobhop cvars.

kiwitf2 opened the pull request:
Adds two cvars based on whats seen with cs:s and csgo (for parity plus it would be VERY nice to have).
Autobunnyhopping also works whilst crouched

"sv_enablebunnyhopping" ( def. "0" )
 game notify replicated
 - Removes speed cap from bunnyhopping.

"sv_autobunnyhopping" ( def. "0")
 game notify replicated
Whether players will automatically bunnyhop whilst holding +JUMP.
   0   - Players will not automatically jump while holding +JUMP.
   1   - Players will automatically jump while holding +JUMP.
   2   - Players will automatically jump while holding +JUMP, and will be able to jump while crouching.

kiwitf2 committed:
bhop cvars

sv_enablebunnyhopping
Removes speed cap from bunnyhopping.

sv_autobunnyhopping
Players will automatically jump while holding space.

both are fcvar_replicated and fcvar_notify
both add bhop to server tags

kiwitf2 committed:
Update tf_gamerules.cpp

remove tags for bhop because yeaj

Bitl commented on src/game/shared/tf/tf_gamemovement.cpp:
```diff
@@ -1237,7 +1243,7 @@ bool CTFGameMovement::CheckJumpButton()
 
 	// Cannot jump again until the jump button has been released.
-	if ( mv->m_nOldButtons & IN_JUMP )
+	if ( mv->m_nOldButtons & IN_JUMP && (!sv_autobunnyhopping.GetBool() || !bOnGround ) )
```
Having !bOnGround in this statement means that this code will break parachutes. After this statement, there's an if statement that checks if scouts can air dash, which is completely skipped when this code is executed. Removing || !bOnGround fixes the issue.

wgetJane commented:
have you tested how this behaves with scout's doublejump?

kiwitf2 committed:
crouching while bhopping can be toggled

if sv_autobunnyhopping is set to 2, players will be able to crouch while bunnyhopping

FlaminSarge commented on src/game/shared/tf/tf_gamemovement.cpp:
Would it not be better to instead have a cvar to change the max speed factor?

kiwitf2 replied:
this is based on how it is in counter strike and im fairly sure there its just enable/disable

FlaminSarge replied:
Is it? I still think it's far more flexible to be able to set the factor by cvar instead.

kiwitf2 replied:
¯\_(ツ)_/¯ i just made this with that in mind, i made this pr a while ago and was and am happy enough with how it is rn tbh, don't really wanna change it all that much

kiwitf2 committed:
make cvar descriptions better match their csgo counterparts
